app/routes/clients.tsx

```diff
@@ -26,13 +29,39 @@ import { analytics, db } from "~/firebase.config";
 import type { Route } from "./+types/clients";
 import { logEvent } from "firebase/analytics";
 
-export const clientLoader = async () => {
+export const clientLoader = async ({ request }: Route.LoaderArgs) => {
```

Comment:
```suggestion
export const clientLoader = async ({ params }: Route.ClientLoaderArgs) => {
```
See https://reactrouter.com/start/framework/data-loading. React Router te fournit déjà les params via l'object passé en argument au `clientLoader`.